x-pack/plugins/siem/server/lib/timeline/routes/utils/compare_timelines_status.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TimelineTypeLiteralWithNull, TimelineType } from '../../../../../common/types/timeline';
+import { FrameworkRequest } from '../../../framework';
+
+import { TimelineStatusActions, TimelineStatusAction } from './common';
+import { TimelineObject } from './timeline_object';
+import {
+  checkIsCreateFailureCases,
+  checkIsUpdateFailureCases,
+  checkIsCreateViaImportFailureCases,
+} from './failure_cases';
+
+interface GivenTimelineInput {
+  id: string | null | undefined;
+  type: TimelineTypeLiteralWithNull;
+  version: string | number | null | undefined;
+}
+
+interface TimelinesStatusProps {
+  timelineType: TimelineTypeLiteralWithNull | undefined;
+  timelineInput: GivenTimelineInput;
+  templateTimelineInput: GivenTimelineInput;
+  frameworkRequest: FrameworkRequest;
+}
+
+export class CompareTimelinesStatus {
+  public readonly timelineObject: TimelineObject;
+  public readonly templateTimelineObject: TimelineObject;
+  private readonly timelineType: TimelineTypeLiteralWithNull;
+
+  constructor({
+    timelineType = TimelineType.default,
+    timelineInput,
+    templateTimelineInput,
+    frameworkRequest,
+  }: TimelinesStatusProps) {
+    this.timelineObject = new TimelineObject({
+      id: timelineInput.id,
+      type: timelineInput.type,
+      version: timelineInput.version,
+      frameworkRequest,
+    });
+
+    this.templateTimelineObject = new TimelineObject({
+      id: templateTimelineInput.id,
+      type: templateTimelineInput.type,
+      version: templateTimelineInput.version,
+      frameworkRequest,
+    });
+
+    this.timelineType = timelineType ?? TimelineType.default;
+  }
+
+  public get isCreatable() {
+    return (
+      (this.timelineObject.isCreatable && !this.isHandlingTemplateTimeline) ||
+      (this.templateTimelineObject.isCreatable &&
+        this.timelineObject.isCreatable &&
+        this.isHandlingTemplateTimeline)
+    );
+  }
+
+  public get isCreatableViaImport() {
+    return this.isCreatable;
+  }
+
+  public get isUpdatable() {
+    return (
+      (this.timelineObject.isUpdatable && !this.isHandlingTemplateTimeline) ||
+      (this.templateTimelineObject.isUpdatable && this.isHandlingTemplateTimeline)
+    );
+  }
+
+  public get isUpdatableViaImport() {
```

Comment:
I am wondering if we are going to have an issue with that
![image](https://user-images.githubusercontent.com/189600/82600327-c1a01d00-9b7b-11ea-8a2f-8db3ab5a23ff.png)